modules/motus/profile/main.nf

```diff
@@ -24,8 +25,9 @@ process MOTUS_PROFILE {
     def prefix = task.ext.prefix ?: "${meta.id}"
     def inputs = "$reads[0]".getExtension == ('.bam') ?
                     "-i ${reads}" :
-                    meta.single_end ?
-                        "-s $reads" : "-f ${reads[0]} -r ${reads[1]}"
+                    meta.mgc ? "-m $reads" :
+                        meta.single_end ?
+                            "-s $reads" : "-f ${reads[0]} -r ${reads[1]}"
```

Comment:
I wouldn't do it like this, we don't want to assume any 'module specific' meta information is in there.

I would suggest replacing `.meta.mqc` with an additional mandatory input channel that specifies either 1) whether input is a mqc file (true/false) or specifies the input file type (bam|fastq|mqc)

Reply:
This issue is resoved by using the file extension.